sdks/python/apache_beam/typehints/typehints.py

```diff
@@ -184,7 +185,17 @@ def bind_type_variables(type_constraint, bindings):
   return type_constraint
 
 
-class SequenceTypeConstraint(TypeConstraint):
+class IndexableTypeConstraint(TypeConstraint):
+  """An internal common base-class for all type constraints with indexing.
+  E.G. SequenceTypeConstraint + Tuple's of fixed size.
+  """
+
+  def _constraint_for_index(self, idx):
```

Comment:
What is this for?

Reply:
Good point, I'll add a docstring. But if you look in simple_type_inference you can see where its used. Because of the insreased use of tuple indexing after conversion it's important that the type inference handle this correctly and the existing code was not.